### PR TITLE
Futurize Python code to make it fully Python 3 compatible

### PR DESCRIPTION
**Pull request by rominf**

### CODE REVIEW

File: compiler/js/component.py [hunk 1]
```diff
@@ -228,7 +231,7 @@ def pregenerate(self, registry):
 		base_type = self.get_base_type(registry, False)
 		base_gen = registry.components[base_type] if base_type != 'core.CoreObject' else None
 
-		for (path, name), (args, code, event, async) in methods.iteritems():
+		for (path, name), (args, code, event, async_) in list(methods.items()):
```
Comment (whoozle) [inline compiler/js/component.py]: here

File: compiler/js/generator.py [hunk 2]
```diff
@@ -110,7 +113,7 @@ def scan_using(self, code):
 
 	def generate_components(self):
 		#finding explicit @using declarations in code
-		for name, code in self.imports.iteritems():
+		for name, code in list(self.imports.items()):
```
Comment (whoozle) [inline compiler/js/generator.py]: here

File: build [hunk 3]
```diff
@@ -282,13 +288,13 @@ def build(root, platform, apps, app, manifest):
 	platform_dirs = []
 	while subplatform_deps:
 		#print subplatform_deps
-		for subplatform, deps in subplatform_deps.items():
+		for subplatform, deps in list(subplatform_deps.items()):
```
[the remark this answers is not in the document]
Comment (rominf) [reply build]: But there is a line:
```python
del subplatform_deps[subplatform]
```
a little bit bellow!

File: build [hunk 4]
```diff
@@ -131,12 +137,12 @@ def process_template_jinja2(destination, source, context):
 		def get_source(self, env, template, updateIndex = False):
 			#print 'get', template, self.paths
 			paths = self.paths
-			for i in xrange(self.next_index.get(template, 0), len(paths)):
+			for i in range(self.next_index.get(template, 0), len(paths)):
 				path = paths[i]
 				src = os.path.join(path, template)
 				if os.path.isfile(src):
-					with open(src) as f:
-						text = f.read().decode('utf-8')
```
[the remark this answers is not in the document]
Comment (rominf) [reply build]: Unfortunately, it's not always possible.

File: compiler/js/component.py (hunk 1 above)
Comment (rominf) [reply compiler/js/component.py]: The list is modified in line:
```python
self.methods[fullname] = args, code, async_
```

Comment (whoozle) [reply compiler/js/component.py]: these are different dictionaries, consider code above
```
        methods = self.methods
        self.methods = {}
```

Comment (rominf) [reply compiler/js/component.py]: Yep, you are right. Sorry.

**Pull request merged**